Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -222,9 +241,23 @@ protected override void ProcessRecord()
 
             // Content type to use for the new object.
             string objContentType = null;
-
             Stream contentStream = null;
-            if (!string.IsNullOrEmpty(File))
+
+            if (ParameterSetName == ParameterSetNames.ContentsFromFolder)
+            {
+                // User gives us the path to a folder, we will resolve the path and upload the contents of that folder.
+                string resolvedFolderPath = GetFullPath(Folder);
+                if (string.IsNullOrWhiteSpace(resolvedFolderPath) || !Directory.Exists(resolvedFolderPath))
+                {
+                    throw new DirectoryNotFoundException($"Directory '{resolvedFolderPath}' cannot be found.");
+                }
+
+                // The relative directory name of this directory will be the name of the directory itself.
+                UploadDirectory(resolvedFolderPath, metadataDict, Path.GetFileName(resolvedFolderPath));
```

Comment:
Do we want the prefix to have to be the folder name, or do we want the user to be able to specify the prefix (logical folder) we want to put the files in?